packages/shared/src/experiments.ts

```diff
@@ -94,3 +103,172 @@ export function getMetricLink(id: string): string {
   if (isFactMetricId(id)) return `/fact-metrics/${id}`;
   return `/metric/${id}`;
 }
+
+export function getRegressionAdjustmentsForMetric<
+  T extends ExperimentMetricInterface
+>({
+  metric,
+  denominatorMetrics,
+  experimentRegressionAdjustmentEnabled,
+  organizationSettings,
+  metricOverrides,
+}: {
+  metric: T;
+  denominatorMetrics: MetricInterface[];
+  experimentRegressionAdjustmentEnabled: boolean;
+  organizationSettings?: Partial<OrganizationSettings>; // can be RA fields from a snapshot of org settings
+  metricOverrides?: MetricOverride[];
+}): {
+  newMetric: T;
+  metricRegressionAdjustmentStatus: MetricRegressionAdjustmentStatus;
+} {
+  const newMetric = cloneDeep<T>(metric);
+
+  // start with default RA settings
+  let regressionAdjustmentEnabled = false;
+  let regressionAdjustmentDays = DEFAULT_REGRESSION_ADJUSTMENT_DAYS;
+  let reason = "";
+
+  // get RA settings from organization
+  if (organizationSettings?.regressionAdjustmentEnabled) {
+    regressionAdjustmentEnabled = true;
+    regressionAdjustmentDays =
+      organizationSettings?.regressionAdjustmentDays ??
+      regressionAdjustmentDays;
+  }
+  if (experimentRegressionAdjustmentEnabled) {
+    regressionAdjustmentEnabled = true;
+  }
+
+  // get RA settings from metric
+  if (metric?.regressionAdjustmentOverride) {
+    regressionAdjustmentEnabled = !!metric?.regressionAdjustmentEnabled;
+    regressionAdjustmentDays =
+      metric?.regressionAdjustmentDays ?? DEFAULT_REGRESSION_ADJUSTMENT_DAYS;
+    if (!regressionAdjustmentEnabled) {
+      reason = "disabled in metric settings";
+    }
+  }
+
+  // get RA settings from metric override
+  if (metricOverrides) {
+    const metricOverride = metricOverrides.find((mo) => mo.id === metric.id);
+    if (metricOverride?.regressionAdjustmentOverride) {
+      regressionAdjustmentEnabled = !!metricOverride?.regressionAdjustmentEnabled;
+      regressionAdjustmentDays =
+        metricOverride?.regressionAdjustmentDays ?? regressionAdjustmentDays;
+      if (!regressionAdjustmentEnabled) {
+        reason = "disabled by metric override";
```

Comment:
Is it possible to have it turned off at the metric and turned off at the override level? It would still work in that case but our reason might get mangled.

Reply:
made the language/reason more clear in this case